lms/services/hapi.py

```diff
@@ -133,8 +134,12 @@ def request(
         except RequestException as err:
             response = getattr(err, "response", None)
             status_code = getattr(response, "status_code", None)
+            if status_code == 404:
+                exception_class = HAPINotFoundError
```

Comment:
Python convention question here: should this var name be capitalized/upper camel case as it's a reference to a class?

Reply:
Since `exception_class` is a local variable that currently references a class, rather than the class itself, lower-case is correct PEP8. We deviate from this rule, I think with good reason, with patch-based test fixtures, but I didn't deviate from it here.